templates/loginform.php

```diff
@@ -297,4 +302,12 @@ class="float-l erroricon" style="margin: 15px "/>
     }
     echo '</script>';
 }
+if ($this->data['doPolling']) {
+    echo '<script type="text/javascript">';
+    foreach ($this->data['pollTokens'] as $i => $e) {
+        SimpleSAML_Logger::debug("Asking client to poll challenges for " . $e . ".");
+        echo 'poll_token_challenges("' . $e . '");';
```

Comment:
Hm, should we maybe escape the token serials to avoid JS injection? I know we validate serials on the server side now, but maybe it's better safe than sorry :) What do you think?

Reply:
I have wrapped this in json_encode(). I *hope* that fixes it.

Reply:
Yes, looks good! Thanks.